### PR TITLE
Reduce vendored cortex code

### PR DESCRIPTION
**Pull request by rabenhorst**
Remove unneeded vendored Cortex code as part of #5912.

### CODE REVIEW

**Comment by GiedriusS** [comment]
Now that's a diff :smile: why is it still a WIP?

**Comment by rabenhorst** [comment]
> Now that's a diff 😄 why is it still a WIP?

It's based on https://github.com/thanos-io/thanos/pull/5896. So I would like to get [this one](https://github.com/thanos-io/thanos/pull/5896.) merged first if possible and then do this PR.

**Comment by rabenhorst** [comment]
> Now that's a diff 😄 why is it still a WIP?

Since https://github.com/thanos-io/thanos/pull/5896 is merged now, I marked this one as read for review.

File: internal/cortex/chunk/chunk.go [hunk 1]
```diff
@@ -353,10 +353,10 @@ func (c *Chunk) Slice(from, through model.Time) (*Chunk, error) {
 	return &nc, nil
 }
 
-func intervalsOverlap(interval1, interval2 model.Interval) bool {
-	if interval1.Start > interval2.End || interval2.Start > interval1.End {
-		return false
-	}
+// func intervalsOverlap(interval1, interval2 model.Interval) bool {
```
Comment (GiedriusS) [inline internal/cortex/chunk/chunk.go:356]: Can we remove this? :smile: why is this commented?

File: internal/cortex/chunk/encoding/factory.go [hunk 2]
```diff
@@ -3,39 +3,34 @@
...
-	return nil
-}
+// // RegisterFlags registers configuration settings.
+// func (Config) RegisterFlags(f *flag.FlagSet) {
+// 	f.Var(&DefaultEncoding, "ingester.chunk-encoding", "Encoding version to use for chunks.")
+// 	f.IntVar(&bigchunkSizeCapBytes, "store.bigchunk-size-cap-bytes", bigchunkSizeCapBytes, "When using bigchunk encoding, start a new bigchunk if over this size (0 = unlimited)")
+// }
+
+// // Validate errors out if the encoding is set to Delta.
+// func (Config) Validate() error {
```
Comment (GiedriusS) [inline internal/cortex/chunk/encoding/factory.go:27]: Same here?

File: internal/cortex/chunk/chunk.go (hunk 1 above)
Comment (rabenhorst) [reply internal/cortex/chunk/chunk.go]: Leftover from my initial attempt of what can be removed. Deleted and rebased on main.

**Review by GiedriusS — APPROVED** [review]
:muscle: less dead code, faster CI. Good work! Also, makes https://github.com/thanos-io/thanos/pull/5421 more feasible because there will be less code to change.

**Pull request merged**